### PR TITLE
Refs #32412 -- Adjusted Contributing Guide start page.

### PR DESCRIPTION
**Pull request by gabbyprecious**
@carltongibson  here are the changes I made, how do they look?

Screenshot from 2021-02-06 15-09-11
Screenshot from 2021-02-06 15-09-21
Screenshot from 2021-02-06 15-09-25

### CODE REVIEW

File: docs/internals/contributing/index.txt [hunk 1]
```diff
@@ -4,30 +4,65 @@ Contributing to Django
...
+    framework yourself.
+
+  * Blog about Django. We syndicate all the Django blogs we know about on
+    the `community page`_; if you'd like to see your blog on that page you
+    can `register it here`_.
+
+  * Contribute to open-source Django projects, write some documentation, or
+    release your own code as an open-source pluggable application. The
+    ecosystem of pluggable applications is a big strength of Django, help us
+    build it!
```
Comment (felixxm) [inline docs/internals/contributing/index.txt:65]: Please remove leading whitespaces:
```suggestion
* Join the `Django forum`_. This forum is a place for discussing the Django
  framework and applications and projects that use it. This is also a good
  place to ask and answer any questions related to installing, using, or
  contributing to Django.

* Join the |django-users| mailing list and answer questions. This
  mailing list has a huge audience, and we really want to maintain a 
  friendly and helpful atmosphere. If you're new to the Django community,
  you should read the `posting guidelines`_.

* Join the `#django IRC channel`_ on Freenode and answer questions. By
  explaining Django to other users, you're going to learn a lot about the
  framework yourself.

* Blog about Django. We syndicate all the Django blogs we know about on
  the `community page`_; if you'd like to see your blog on that page you
  can `register it here`_.

* Contribute to open-source Django projects, write some documentation, or
  release your own code as an open-source pluggable application. The
  ecosystem of pluggable applications is a big strength of Django, help us
  build it!
```

File: docs/internals/contributing/index.txt [hunk 2]
```diff
@@ -4,30 +4,65 @@ Contributing to Django
...
-  framework and applications and projects that use it. This is also a good
-  place to ask and answer any questions related to installing, using, or
-  contributing to Django.
+There are three major ways to contribute to Django:
 
-* Join the |django-users| mailing list and answer questions. This
-  mailing list has a huge audience, and we really want to maintain a
-  friendly and helpful atmosphere. If you're new to the Django community,
-  you should read the `posting guidelines`_.
+  * **Coding**
```
Comment (felixxm) [inline docs/internals/contributing/index.txt:11]: The same here:
```suggestion
* **Coding**
```

File: docs/internals/contributing/index.txt [hunk 3]
```diff
@@ -4,30 +4,65 @@ Contributing to Django
 
 Django is a community that lives on its volunteers. As it keeps growing, we
 always need more people to help others. As soon as you learn Django, you can
-contribute in many ways:
+contribute in many ways.
```
Comment (felixxm) [inline docs/internals/contributing/index.txt:7]: Remove trailing whitespaces.
```suggestion
contribute in many ways.
```

File: docs/internals/contributing/index.txt [hunk 4]
```diff
@@ -46,27 +81,11 @@ development:
   `easy pickings`_ tickets. The :ref:`patch-review-checklist` will also be
   helpful.
 
-* :doc:`Improve the documentation <writing-documentation>` or
-  :doc:`write unit tests <writing-code/unit-tests>`.
+* :doc:`write unit tests <writing-code/unit-tests>`.
```
Comment (felixxm) [inline docs/internals/contributing/index.txt:84]:
```suggestion
* :doc:`Write unit tests <writing-code/unit-tests>`.
```

**Review by carltongibson — COMMENTED** [review]
Hey @gabbyprecious. 

I think this is good. 

* I rebased to remove the merge commits. Please either use `git reset` to get fbb0ae5fce73b128b4abd2d0777feb020894c70b here, and add on top, or delete your local branch and create a new branch again from the remote one here. (This amounts to the same thing 😀)

Then have a look at fbb0ae5fce73b128b4abd2d0777feb020894c70b for edits I added. See the commit message there for thoughts on it. 

1. I want to have a think about the text for the top three main points here, but... 
2. The extra bits I removed are perfect I think (adjusted) for the `writing-code/index` page, that we've talked about. 

Let me know what you think, and I'll tidy this up for merge. (Do that wording, squash and commit message format, etc). 

Thanks!

**Review by smithdc1 — COMMENTED** [review]
This is great :clap: 

I've left a few writing style comments and as such, they are very subjective. 

I think the main thing to call out is that I think the community engagement piece should feature more prominently. Contributing to the wider community/ecosystem is a great way to get started and is very much valued (some folk on the forum making outstanding contributions). I therefore think it's a bit of a missed opportunity as currently presented. 

(I've almost certainly got some unconscious bias in advocating this ecosystem-first approach so please do challenge and correct me here!).

**Comment by gabbyprecious** [comment]
@carltongibson I've reset my brnch locally and made some changes based on @smithdc1 suggestions. Have a look and merge

**Comment by gabbyprecious** [comment]
@carltongibson this is yet to be merged and I think we're done on the review

**Comment by carltongibson** [comment]
Yes, hi @gabbyprecious, sorry about that. We had security releases last week, which took over, and I’m on holiday this week. I will get to it as soon as I’m back. Thanks!

**Review by carltongibson — APPROVED** [review]
Hi @gabbyprecious — Thanks again for this. I think it makes quite a big difference to the approachability of the Contributing Guide. 

Some points: 

* I've made a few working adjustments to the top-three callouts.
* I added headers for Working on the framework vs Join the community (plus adjusted the leading sentence. Hopefully this address @smithdc1's point in giving the Community side the weight it needs. (Note the headers mean the sections are called out in the page ToC in the sidebar.) 
* I **moved** the Writing code specific points (which I'd marked for deletion) to the writing code index page. With the new starting callouts, this makes it much clearer I think, and adds the (start of the) needed content for that Writing code section. (That page looks much less flimsy now.) 
* I then squashed, rebased, and used the project commit message format and so on. 

Good stuff 👍

**Comment by carltongibson** [comment]
Note to self maybe: With this in place I have one wish to adjust the Advice to New Contributors, where I think Triaging Tickets perhaps needs replacing with something about finding a ticket, and spending some time with it (that's been discussed in a few place, if you were there for that). Other than that, things can always be tweaked but I'm quite please with how this has come out... (Thanks @gabbyprecious!)

File: docs/internals/contributing/index.txt [hunk 5]
```diff
@@ -3,8 +3,59 @@ Contributing to Django
...
+    Start from the :doc:`writing-code/index` docs.
+
+**Writing documentation**
+    Django's documentation is one of its key strengths. It's informative
+    and thorough. You can help to improve the documentation and keep it
+    relevant as the framework evolves.
+
+    See :doc:`writing-documentation` for more.
+
+**Localizing Django**
```
[the remark this answers is not in the document]
Comment (felixxm) [reply docs/internals/contributing/index.txt:29]: Maybe a few more emojis :smile:

File: docs/internals/contributing/index.txt [hunk 6]
```diff
@@ -3,8 +3,59 @@ Contributing to Django
...
+
+The work on Django itself falls into three major areas:
+
+**Writing code**
+    Fix a bug, or add a new feature. You can make a pull request and see **your
+    code** in the next version of Django!
+
+    Start from the :doc:`writing-code/index` docs.
+
+**Writing documentation**
```
Comment (felixxm) [inline docs/internals/contributing/index.txt:22]:
```suggestion
**Writing documentation** ✍️
```

File: docs/internals/contributing/index.txt [hunk 7]
```diff
@@ -3,8 +3,59 @@ Contributing to Django
...
+
+As soon as you learn Django, you can contribute in many ways, either on the
+framework itself or in the wider ecosystem.
+
+Work on the Django framework
+============================
+
+The work on Django itself falls into three major areas:
+
+**Writing code**
```
Comment (felixxm) [inline docs/internals/contributing/index.txt:16]:
```suggestion
**Writing code** 💻
```

File: docs/internals/contributing/index.txt [hunk 8]
```diff
@@ -3,8 +3,59 @@ Contributing to Django
...
+   new-contributors
+   bugs-and-features
+   triaging-tickets
+   writing-code/index
+   writing-documentation
+   localizing
+   committing-code
+
+Join the Django community
+=========================
```
Comment (felixxm) [inline docs/internals/contributing/index.txt:54]:
```suggestion
Join the Django community ❤️
============================
```

**Review by felixxm — APPROVED** [review]
Thanks y'all :rocket: This looks great :1st_place_medal: 

I only left some minor suggestions that can be ignored.

**Comment by carltongibson** [comment]
Yep, let's have all those emojis 😀

**Comment by smithdc1** [comment]
Sorry, was at work today so only getting to look at this now. 

This looks great, super work @gabbyprecious!

I did have a good chuckle when I read the note about Klingon! 🙂

I like the addition of the emojis! 🎉 Only question is do they go through the translation process ok?

**Comment by carltongibson** [comment]
Surely they do 😀.  @claudep will let us know if it breaks anything 🙂

**Comment by claudep** [comment]
I'll only be able to tell when this will reach the real translation workflow in some months! We'll see how Transifex handle those, I guess this mostly depend on the font used to edit the translation content.

**Comment by carltongibson** [comment]
OK, thanks @claudep. 🤞— we can adjust if a problem does come up.

**Pull request merged**